[Feature] 게시판 생성시 카테고리 동적 추가, 게시판의 카테고리 조회 기능 구현

u-genuine opened the pull request:
---
name: PULL_REQUEST_TEMPLATE
about: Describe this issue template's purpose here.
title: ''
labels: ''
assignees: ''

---

### PR Type
<!— Please check the one that applies to this PR using "x". —>

- [ ] 버그수정(Bugfix)
- [x] 기능개발(Feature)
- [ ] 코드 스타일 변경(Code style update) (formatting, local variables)
- [x] 리팩토링 (no functional changes, no api changes)
- [ ] 빌드 관련 변경
- [ ] 문서 내용 변경
- [ ] Other… Please describe:

### 요약(Summary)
- 게시판 생성API 카테고리 추가 구현
- 게시판 생성 API 테스트 코드 추가

### 상세 내용(Describe your changes)
#### 게시판 생성API 카테고리 추가 구현
<img width="919" alt="image" src="https://github.com/user-attachments/assets/157ca6a7-1abc-4362-93f6-2e175482a28b" />

**Request**
- isCategoryRequired 제거 -> List<String> categories 받도록 수정
- universityName(한글 대학명) -> universityCode(영문 대학명) 수정

**Entity**
- Category, PostCategory 엔티티 생성

#### 게시판에 적용되는 카테고리 조회 기능 구현
<img width="925" alt="image" src="https://github.com/user-attachments/assets/56c97023-f495-4ac9-9595-911a7f591474" />

```
{
    "status": "OK",
    "data": {
        "categories": [
            "A",
            "B",
            "C"
        ]
    },
    "timestamp": "2025-03-18 20:30:02"
}
```

```
{
    "code": "BOARD-001",
    "message": "해당 게시판을 찾을 수 없습니다.",
    "method": "GET",
    "requestURI": "/v1/api/posts/boards/2/categories"
}

```

### Issue Number or Link
#179

u-genuine committed:
Test: UserService 단위테스트 추가

u-genuine committed:
Feature: Category, PostCategory 엔티티 추가

u-genuine committed:
Refactor: 대학 게시판 생성시 대학 영문명 받도록 수정

u-genuine committed:
Refactor: Board 엔티티의 isCategoryRequired 필드명을 usesCategories로 변경

u-genuine committed:
Refactor: Board 엔티티의 isCategoryRequired 필드명을 usesCategories로 변경

u-genuine committed:
Feature: 게시판 생성 API - 카테고리 동적 생성 기능 구현

u-genuine committed:
Docs: API 설명 수정

u-genuine committed:
Test: createBoard 테스트코드 추가

u-genuine committed:
Merge remote-tracking branch 'origin/develop' into feature/173

u-genuine committed:
Merge remote-tracking branch 'origin/develop' into feature/173

u-genuine committed:
Feature: 게시판에 적용되는 카테고리 조회 API 구현

u-genuine committed:
Chore: .gitignore 파일 업데이트

goalSetter09 commented on src/main/java/com/cotato/kampus/domain/admin/dto/AdminBoardDetail.java:
```diff
@@ -20,7 +20,7 @@ public static AdminBoardDetail of(BoardDto boardDto, Long postCount, Long deleti
 			boardDto.universityId(),
 			boardDto.description(),
-			boardDto.isCategoryRequired(),
+			boardDto.usesCategories(),
```
이거 이름이 바뀐 이유가 뭘까용?

u-genuine replied:
카테고리가 있는 게시판에 글을 쓸 때 카테고리 설정이 필수가 아니여서 바꿨습니다!

goalSetter09 commented on src/main/java/com/cotato/kampus/domain/admin/dto/request/BoardCreateRequest.java:
```diff
@@ -1,14 +1,14 @@
 import jakarta.validation.constraints.NotBlank;
-import jakarta.validation.constraints.NotNull;
 
 public record BoardCreateRequest(
```
다른 필드에도 제약조건 추가해주시면 좋을 것 같습니다!

u-genuine replied:
엇 필수인지 확인해보고 추가하도록 하겠습니다!

goalSetter09 commented on src/main/java/com/cotato/kampus/domain/board/application/CategoryAppender.java:
```diff
@@ -0,0 +1,31 @@
+@Component
+@Transactional(readOnly = true)
+@RequiredArgsConstructor(access = AccessLevel.PROTECTED)
+public class CategoryAppender {
```
한 칸 띄우면 좋을 것 같습니다!

u-genuine replied:
네 수정하겠습니다!

goalSetter09 commented on src/main/java/com/cotato/kampus/domain/board/domain/Board.java:
```diff
@@ -40,8 +38,8 @@ public class Board extends BaseTimeEntity {
 
-	@Column(name = "is_category_required", nullable = false)
-	private Boolean isCategoryRequired;
+	@Column(name = "uses_Categories", nullable = false)
```
컬럼 이름은 소문자로 통일해주시면 좋겠습니다!

u-genuine replied:
헉 죄송해요 꼼꼼하게 봐주셔서 감사합니당

goalSetter09 commented on src/main/java/com/cotato/kampus/domain/board/domain/Category.java:
```diff
@@ -0,0 +1,36 @@
+@Entity
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class Category extends BaseTimeEntity {
```
클래스 이름을 카테고리로 하면, 추후에 다른 곳에서 또 카테고리가 사용되는 경우에 문제가 생길 것 같아요!

u-genuine replied:
아 그러네요..! BoardCategory로 수정하겠습니다

goalSetter09 commented on src/test/java/com/cotato/kampus/domain/board/application/BoardServiceTest.java:
```diff
@@ -0,0 +1,128 @@
+	private BoardService boardService;
+
+	@Test
+	void 존재하는_게시판의_카테고리_목록을_조회() {
```
여기서도 @DisplayName을 통해서 한글 설명을 추가하고, 매서드 이름은 영어로 유지하는 걸로 하면 어떨까요?!

goalSetter09 reviewed (commented):
오.. 테스트 코드 다 작성하느라 고생하셨습니다..!!!

goalSetter09 commented:
아 그리고 혹시 커밋 메시지에 feat(#179): ~~~
이런식으로 이슈 넘버 표시해주실 수 있나요?! 어떤 이슈랑 관련된 커밋인지 보기 편할 것 같아요!
앞으로 해주시는 커밋에만 적용해주셔도 괜찮습니다..

goalSetter09 commented on .gitignore:
```diff
@@ -19,7 +19,7 @@ bin/
 ### IntelliJ IDEA ###
 application.yml
-application-prod.yml
+application-test.yml
```
엇 이거 prod도 있어서 빼시면 안될 것 같습니다..!

u-genuine committed:
Refactor(IT-Cotato#179): 코드 리뷰 반영
- .gitignore 파일 수정
- 코드 줄바꿈
- Board 엔티티 컬럼명 소문자로 수정
- Category -> BoardCategory 엔티티명 변경
- 테스트 메소드 이름 수정 및 @DisplayName 추가

u-genuine commented:
리뷰 반영해서 수정했습니다! 꼼꼼하게 봐주셔서 감사합니다 :)
